artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/LinkedListIterator.java

```diff
@@ -27,6 +27,9 @@ public interface LinkedListIterator<E> extends Iterator<E>, AutoCloseable {
 
    void repeat();
 
+   /** This method is doing exactly what {@link Iterator#remove()} would do, however it will return the removed element being removed. */
+   E removeLastElement();
```

Comment:
removeLastElement() seems ambiguous. If its the same as remove(), which operates on the _current_ element in most Iterators, something like removeCurrentElement() element would probably be more straightforward for later users. Or something much more on the nose like removeAndReturn().